drivers/gpio/gpio_stm32.c

```diff
@@ -585,57 +587,57 @@ static int gpio_stm32_init(struct device *device)
 			    CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,	       \
 			    &gpio_stm32_driver)
 
-#define GPIO_DEVICE_INIT_STM32(__suffix, __SUFFIX)		      \
-	GPIO_DEVICE_INIT(DT_GPIO_STM32_GPIO##__SUFFIX##_LABEL,	      \
-			 __suffix,				      \
-			 DT_GPIO_STM32_GPIO##__SUFFIX##_BASE_ADDRESS, \
-			 STM32_PORT##__SUFFIX,			      \
-			 DT_GPIO_STM32_GPIO##__SUFFIX##_CLOCK_BITS,   \
-			 DT_GPIO_STM32_GPIO##__SUFFIX##_CLOCK_BUS)
+#define GPIO_DEVICE_INIT_STM32(__suffix, __SUFFIX)			\
+	GPIO_DEVICE_INIT(DT_LABEL(DT_NODELABEL(gpio##__suffix)),	\
+			 __suffix,					\
+			 DT_REG_ADDR(DT_NODELABEL(gpio##__suffix)),	\
+			 STM32_PORT##__SUFFIX,				\
+			 DT_CLOCKS_CELL(DT_NODELABEL(gpio##__suffix), bits),\
+			 DT_CLOCKS_CELL(DT_NODELABEL(gpio##__suffix), bus))
 
-#ifdef CONFIG_GPIO_STM32_PORTA
```

Comment:
For the driver is there a reason it can't use DT_INST form, doesn't seem like there is any need/use for DT_NODELABEL here.

Reply:
Using DT_INST would not work with the driver as is.
In the `GPIO_DEVICE_INIT_STM32` macro, we're building a constant name (`STM32_PORT##__SUFFIX`).
This constant is later used in this driver and as well interrupt controller and pinmux bindings.

So we should be able to get rid of that, but this is not immediate.